### PR TITLE
Calico swarm tutorial

### PR DESCRIPTION
**Pull request by caseydavenport**
Take a look at it here:  https://github.com/Metaswitch/calico-docker/blob/calico-swarm-tutorial/docs/CalicoSwarm.md

### CODE REVIEW

File: docs/CalicoSwarm.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Running calico-docker networking on a Docker Swarm.
+The following instructions configure a Docker Swarm cluster networked using Calico.  In this tutorial, we will do the following.
+- Configure Docker Swarm on a VM cluster.
+- Configure etcd and Calico on our cluster.
+- Provision workloads on the Swarm, and check they can communicate.
+
+## Prerequisites
+To complete this demo, make sure you have the following prerequisites.
+- Four Linux servers (VMs or bare-metal) with Docker 1.4 or later installed.  The servers should have IP connectivity to each other.
+  - 1 client (to control the Swarm)
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: Should this be Swarm client to be in keeping with the next two lines.

Comment (robbrockbank) [inline docs/CalicoSwarm.md]: I'm assuming the client, swarm manager and swarm nodes are not pre-requisites and that the instructions below tell you how to install them?

File: docs/CalicoSwarm.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Now, let's create some containers on our cluster. Run the following commands on your client node.
+```
+docker -H $MANAGER_IP:$SWARM_PORT run -e CALICO_IP=192.168.1.1 CALICO_PROFILE=PROF_A_B_C --name workload-A -tid busybox
+docker -H $MANAGER_IP:$SWARM_PORT run -e CALICO_IP=192.168.1.2 CALICO_PROFILE=PROF_A_B_C --name workload-B -tid busybox
+docker -H $MANAGER_IP:$SWARM_PORT run -e CALICO_IP=192.168.1.3 CALICO_PROFILE=PROF_A_B_C --name workload-C -tid busybox
+docker -H $MANAGER_IP:$SWARM_PORT run -e CALICO_IP=192.168.1.4 CALICO_PROFILE=PROF_D --name workload-D -tid busybox
+docker -H $MANAGER_IP:$SWARM_PORT run -e CALICO_IP=192.168.1.5 CALICO_PROFILE=PROF_E --name workload-E -tid busybox
+```
+
+We can run ```ps``` against the Swarm manager to check that the containers haveb been created.
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: haveb -> have

File: docs/CalicoSwarm.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  - 1 client (to control the Swarm)
+  - 1 Swarm manager (to manage the Swarm nodes)
+  - 2 Swarm nodes (to host our containers)
+
+## Installing Swarm on your cluster
+Our Docker Swarm will consist of 2 nodes and a Swarm manager.  We will use the fourth server as the client to control our Swarm.  We'll use the token-based discovery backend to configure our Swarm.
+
+First, let us create the token which will identify our swarm.  To do this, run the following on your client node:
+```
+docker pull swarm
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: Might be worth indicating that sudo could be required for the docker commands.  Depends on your docker installation.

File: docs/CalicoSwarm.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+The second command will print a unique cluster ID token which we must use to configure our Swarm.  We'll reference this value in the tutorial as ```<swarm_token>```. 
+Your token should look something like this: d435dc114a3dd89af2fae239257338ce
+
+Now that we've got a token, we can begin to configure our Swarm.  First, let's join each of our two Swarm nodes to the
+cluster.  Run the following commands on each node, replacing ```<swarm_token>``` with the token from above, and ```<node_ip>``` with the IP address of this node. 
+```
+docker run -d swarm join --addr=<node_ip>:2377 token://<swarm_token>
+```
+
+Let's now configure the Swarm manager node.  To do this, run the following on your Swarm manager node.  Note that ```<swarm_port>``` in the following command can be any unused TCP port on the manager server.  This is the port the client will use to communicate with the Swarm manager daemon.
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: Worth saying how to choose an unused port?

File: docs/CalicoSwarm.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+node, so store it in an environment variable for easy access.
+```
+export MANAGER_IP=<Manager's IP address>
+```
+
+Use the following command to start etcd:
+```
+etcd -name etcd0  -advertise-client-urls http://$MANAGER_IP:2379,http://$MANAGER_IP:4001  -listen-client-urls http://0.0.0.0:2379,http://0.0.0.0:4001  -initial-advertise-peer-urls http://$MANAGER_IP:2380  -listen-peer-urls http://0.0.0.0:2380  -initial-cluster-token etcd-cluster-1  -initial-cluster etcd0=http://$MANAGER_IP:2380 -initial-cluster-state new &
+
+```
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: Worth following up with a simple ps to verify etcd is running?  or some other way?

File: docs/CalicoSwarm.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+To make things simpler, let's store a couple of commonly used values in environment variables.
+```
+export MANAGER_IP=<Manager Node's IP Address>
+export NODE_IP=<This Node's IP Address>
+```
+
+These commands download and start Calico on the node.
+```
+# Download calicoctl and make it executable:
+wget https://github.com/Metaswitch/calico-docker/releases/download/v0.4.5/calicoctl
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: May need to update to the latest binary

File: docs/CalicoSwarm.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+# Point this node at the etcd cluster
+export ETCD_AUTHORITY=$MANAGER_IP:4001
+
+# Configure local Docker requests to be routed through Powerstrip.
+export DOCKER_HOST=localhost:2377
+
+# Start the calico node service.  We must specify the ETCD_AUTHORITY variable since sudo uses its own environment.
+sudo ETCD_AUTHORITY=$MANAGER_IP:4001 ./calicoctl node --ip=$NODE_IP
+```
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: Run docker ps to check that it is running.

File: docs/CalicoSwarm.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+docker run --rm swarm create
+``` 
+
+The second command will print a unique cluster ID token which we must use to configure our Swarm.  We'll reference this value in the tutorial as ```<swarm_token>```. 
+Your token should look something like this: d435dc114a3dd89af2fae239257338ce
+
+Now that we've got a token, we can begin to configure our Swarm.  First, let's join each of our two Swarm nodes to the
+cluster.  Run the following commands on each node, replacing ```<swarm_token>``` with the token from above, and ```<node_ip>``` with the IP address of this node. 
+```
+docker run -d swarm join --addr=<node_ip>:2377 token://<swarm_token>
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: Might be nice to setup $NODE_IP beforehand to use this.  Maybe set up all the env variables upfront.

File: docs/CalicoSwarm.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+```
+# Set the IP of the Swarm manager node
+export MANAGER_IP=<Manager Node's IP Address>
+
+# Download calicoctl and make it executable:
+wget https://github.com/Metaswitch/calico-docker/releases/download/v0.4.5/calicoctl
+chmod +x ./calicoctl
+
+# Point this node at the etcd cluster
+ETCD_AUTHORITY=$MANAGER_IP:4001
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: export?

File: docs/CalicoSwarm.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+chmod +x ./calicoctl
+
+# Point this node at the etcd cluster
+export ETCD_AUTHORITY=$MANAGER_IP:4001
+
+# Configure local Docker requests to be routed through Powerstrip.
+export DOCKER_HOST=localhost:2377
+
+# Start the calico node service.  We must specify the ETCD_AUTHORITY variable since sudo uses its own environment.
+sudo ETCD_AUTHORITY=$MANAGER_IP:4001 ./calicoctl node --ip=$NODE_IP
```
Comment (robbrockbank) [inline docs/CalicoSwarm.md]: You might need to do a checksystem --fix first to ensure the correct kernel modules are loaded.

File: docs/CalicoSwarm.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+- Configure Docker Swarm on a VM cluster.
+- Configure etcd and Calico on our cluster.
+- Provision workloads on the Swarm, and check they can communicate.
+
+## Prerequisites
+To complete this demo, make sure you have the following prerequisites.
+- Four Linux servers (VMs or bare-metal) with Docker 1.4 or later installed.  The servers should have IP connectivity to each other.
+  - 1 client (to control the Swarm)
+  - 1 Swarm manager (to manage the Swarm nodes)
+  - 2 Swarm nodes (to host our containers)
```
Comment (lxpollitt) [inline docs/CalicoSwarm.md]: These should be 2 and 3.

File: docs/CalicoSwarm.md (hunk 1 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Yep - changed.

File: docs/CalicoSwarm.md (hunk 3 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Yep, done.

File: docs/CalicoSwarm.md (hunk 6 above)
Comment (lxpollitt) [inline docs/CalicoSwarm.md]: Yes, 0.4.6.

File: docs/CalicoSwarm.md (hunk 7 above)
Comment (lxpollitt) [inline docs/CalicoSwarm.md]: No, I don't think this is worth doing.  We aren't this defensive in other tutorials or demos (even the live demos we give).

File: docs/CalicoSwarm.md (hunk 5 above)
Comment (lxpollitt) [inline docs/CalicoSwarm.md]: I don't think this is worth doing.

File: docs/CalicoSwarm.md (hunk 6 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: This demo doesn't rely on anything in the latest version, and **may** be broken by libnetwork changes, I'm not sure yet.  I'll leave it as-is until verified.

File: docs/CalicoSwarm.md (hunk 8 above)
Comment (lxpollitt) [inline docs/CalicoSwarm.md]: Not sure that is worth doing.  It looks to me like the swarm_token is only used once on each node.  What's the point in setting it in an env variable?

File: docs/CalicoSwarm.md (hunk 10 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Yep, done.

File: docs/CalicoSwarm.md (hunk 7 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Yep, done.

File: docs/CalicoSwarm.md (hunk 9 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Oops, fixed.

File: docs/CalicoSwarm.md (hunk 2 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Fixed.

File: docs/CalicoSwarm.md (hunk 8 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: Yep, agreed with lxp

File: docs/CalicoSwarm.md (hunk 11 above)
Comment (caseydavenport) [inline docs/CalicoSwarm.md]: They were meant to be indications of how many of each node type we're using, rather than a numbered list.

**Pull request closed without merging**